test/IXWebSocketPingTest.cpp

```diff
@@ -188,36 +192,161 @@ TEST_CASE("Websocket_heartbeat", "[heartbeat]")
 
         // Wait for all chat instance to be ready
         while (true)
         {
-            if (webSocketClientA.isReady() && webSocketClientB.isReady()) break;
+            if (webSocketClient.isReady()) break;
             ix::msleep(10);
         }
 
-        REQUIRE(server.getClients().size() == 2);
+        REQUIRE(server.getClients().size() == 1);
+
+        ix::msleep(1900);
+
+        webSocketClient.stop();
+
+
+        // Here we test ping interval 
+        // -> expected ping messages == 1 as 1900 seconds, 1 ping sent every second
+        REQUIRE(serverReceivedPingMessages == 1);
+
+        // Give us 500ms for the server to notice that clients went away
+        ix::msleep(500);
+        REQUIRE(server.getClients().size() == 0);
+
+        ix::reportWebSocketTraffic();
+    }
+}
+
+TEST_CASE("Websocket_ping_data_sent_setHeartBeatPeriod", "[setHeartBeatPeriod]")
+{
+    SECTION("Make sure that ping messages are sent, even if other messages are sent")
+    {
+        ix::setupWebSocketTrafficTrackerCallback();
+
+        int port = getFreePort();
+        ix::WebSocketServer server(port);
+        std::atomic<int> serverReceivedPingMessages(0);
+        REQUIRE(startServer(server, serverReceivedPingMessages));
+
+        std::string session = ix::generateSessionId();
+        WebSocketClient webSocketClient(port, true); // true to use setHeartBeatPeriod
+
+        webSocketClient.start();
+
+        // Wait for all chat instance to be ready
+        while (true)
+        {
+            if (webSocketClient.isReady()) break;
+            ix::msleep(10);
+        }
+
+        REQUIRE(server.getClients().size() == 1);
 
         ix::msleep(900);
-        webSocketClientB.sendMessage("hello world");
-        ix::msleep(900);
-        webSocketClientB.sendMessage("hello world");
+        webSocketClient.sendMessage("hello world");
         ix::msleep(900);
+        webSocketClient.sendMessage("hello world");
+        ix::msleep(1100);
+
+        webSocketClient.stop();
+
+        // Here we test ping interval 
+        // client has sent data, but ping should have been sent no matter what
+        // -> expected ping messages == 2 as 900+900+1100 = 2900 seconds, 1 ping sent every second
+        REQUIRE(serverReceivedPingMessages == 2);
+
+        // Give us 500ms for the server to notice that clients went away
+        ix::msleep(500);
+        REQUIRE(server.getClients().size() == 0);
+
+        ix::reportWebSocketTraffic();
+    }
+}
+
+TEST_CASE("Websocket_ping_no_data_sent_setPingInterval", "[setPingInterval]")
+{
+    SECTION("Make sure that ping messages are sent when no other data are sent.")
+    {
+        ix::setupWebSocketTrafficTrackerCallback();
+
+        int port = getFreePort();
+        ix::WebSocketServer server(port);
+        std::atomic<int> serverReceivedPingMessages(0);
+        REQUIRE(startServer(server, serverReceivedPingMessages));
+
+        std::string session = ix::generateSessionId();
```

Comment:
Sometimes I create a temporary variable here to self comment things, instead of using 'false' which we don't know what that false is for.

Reply:
There are few test case where I put true or false with a comment explaining, but we could create a var locally too